doc/user/formats.rst

```diff
@@ -130,3 +130,71 @@ fractional_ranges
         be masked if :math:`[\frac{i-0.5}{nchans}, \frac{i+0.5}{nchans}]`
         overlaps any of the ranges. Note that this means that channel 0
         is centred at 0.0 but extends below it.
+
+Primary beam
+------------
+The file contains the information in a compact Fourier-transform ("aperture
+plane") form. It comprises a 3-dimensional array J of Jones matrices. Another
+three arrays :math:`\nu`, :math:`y` and :math:`x` indicate the frequency (in
+Hz) and spatial position (in metres) of each sample along the axes.
+
+To determine the response at a sampled frequency :math:`\nu_f` and some
+direction, turn the direction into direction cosines :math:`l` and :math:`m`
+relative to the pointing centre (as defined in
+:class:`katsdpmodels.primary_beam.AltAzFrame`). Then the Jones
+matrix for the response is
+
+.. math:: \frac{1}{|x|\cdot|y|}
+          \sum_{j,k} e^{-2\pi i (x_j l + y_k m)\nu_f/c} J_{f,k,j},
+
+where :math:`|x|`, :math:`|y|` are the number of elements in the respective
+arrays and :math:`c` is the speed of light.
+The Jones matrices correspond to :attr:`.OutputType.JONES_HV`. Note the
+reversed axis order in accessing :math:`J`.
+
+To sample at an intermediate frequency, use linear interpolation along the
+frequency axis in the aperture plane.
+
+See :class:`katsdpmodels.primary_beam.PrimaryBeam` for other definitions and
+sign conventions.
+
+Attributes
+^^^^^^^^^^
+model_type
+    ``primary_beam``
+
+model_format
+    ``aperture_plane``
+
+antenna
+    The name of the antenna to which this model applies. Absent if this model
+    is not specific to a single antenna (the more generic ``model_target`` may
+    provide human-readable information about the range of applicable antennas).
+
+receiver
+    Serial number of the receiver to which this model applies. Absent if this
+    model is not specific to a single receiver.
+
+x_start, y_start
+    Coordinates associated with the first sample along the respective axes.
+
+y_step, y_step
+    Spacing between samples along the respective axes.
+
+The `antenna` and `receiver` may be compared to the actual antenna
+and receiver identifiers in use to detect incorrect models (for example, if a
```

Comment:
I think about it the opposite way... Start with the actual antenna + receiver and find the most appropriate model in the store (I guess via aliases).

Back off to a more generic model and reprocess with a more appropriate model once you have it.

If the model is "incorrect", then what do you do? Complain? Refuse to run the pipeline? This is probably my biggest worry about katsdpmodels - who is going to maintain the whole edifice?

Reply:
Doing lookup with both antenna and receiver is an interesting idea. While the antenna+band should be sufficient to specify a current model (on the assumption that we won't put two receivers of the same band on the same antenna), Mattieu's research might in future let us predict new combinations, and the sdpmodels server could then synthesize a model on the fly (assuming both the dish and the receiver had been characterised separately). On the other hand, it means we have to care about receiver serial numbers even if we don't use dish-specific models, so I'm reluctant to make it a starting point.

One question is whether, if we only have a model for a dish with a different receiver, that is still better than using the array average (because at least we're capturing dish surface errors), or worse (because we're assuming completely wrong collimation errors). @mattieudv any thoughts?

> If the model is "incorrect", then what do you do? Complain? Refuse to run the pipeline?

This is something I've struggled with, and it probably again depends on how much we depend on per-dish models to get the quality we need. We probably need to squawk loudly enough that someone will go update the model, but we probably can't refuse to run because otherwise there would be no way to do the holography observations needed to update the model.

At this point I'm considering adding it mostly because it seems like it might be useful to have in future, rather than having a definite plan in mind.

Reply:
Yes, many non public domain thoughts.

Reply:
But you should allow for receiver specification in the future yes.